pandas/core/groupby/groupby.py

```diff
@@ -2550,9 +2550,8 @@ def _get_cythonized_result(
         """
         if result_is_index and aggregate:
             raise ValueError("'result_is_index' and 'aggregate' cannot both be True!")
-        if post_processing:
-            if not callable(post_processing):
-                raise ValueError("'post_processing' must be a callable!")
+        if post_processing and not callable(post_processing):
+            raise ValueError("'post_processing' must be a callable!")
         if pre_processing:
             if not callable(pre_processing):
                 raise ValueError("'pre_processing' must be a callable!")
```

Comment:
Could use here ``if pre_processing and not callable(pre_processing)`` to make it similar to the other ones.

Reply:
Are you sure we'd want that? Because here the `if pre_processing` is shared with the following check, so the change would become
```diff
-        if pre_processing:
-            if not callable(pre_processing):
-                raise ValueError("'pre_processing' must be a callable!")
-            if not needs_values:
-                raise ValueError(
-                    "Cannot use 'pre_processing' without specifying 'needs_values'!"
-                )
+        if pre_processing and not callable(pre_processing):
+            raise ValueError("'pre_processing' must be a callable!")
+        if pre_processing and not needs_values:
+            raise ValueError(
+                "Cannot use 'pre_processing' without specifying 'needs_values'!"
+            )
```
which IMO wouldn't be an improvement, but I'm happy to make it if it you think it is

Reply:
My bad. I did not look through the whole if statement below.